File: realworld-back/realworld-model/src/main/java/com/realworld/entity/Article.java
package com.realworld.entity;

import com.baomidou.mybatisplus.annotation.*;

import java.time.LocalDateTime;
import java.io.Serializable;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * 文章表
 * </p>
 *
 * @author 
 * @since 2025-04-18
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("article")
@NoArgsConstructor
@Schema(description="文章表")
public class Article implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @Schema(description = "文章标题")
    private String title;

    @Schema(description = "文章简介")
    private String description;

    @Schema(description = "文章内容")
    private String body;

    @Schema(description = "作者id")
    private Integer authorId;

    @Schema(description = "点赞数量")
    private int likeCount;

    @Schema(description = "收藏数量")
    private int favoritesCount;

    @Schema(description = "评论数量")
    private int commentCount;

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @Schema(description = "标记删除状态")
    private int isDel;


}

File: front/src/types/response/article.ts
import type {ProfileRes} from "@/types/response/user.ts";

export interface ArticleRes {
  id: string;
  title: string;
  description: string;
  body: string
  tags: string
  createdAt: string
  liked: boolean
  favorited: boolean
  favoritesCount: number
  commentCount: number
  likeCount: number
  author: ProfileRes
}

export interface ArticleCardRes {
  id: string;
  title: string;
  description: string;
  tags: string
  createdAt: string
  likeCount: number;
  favoritesCount: number;
  liked: boolean;
  favorited: boolean;
  author: string;
  avatar: string;
}
